apps/evm/src/clients/api/queries/getImportablePositions/getImportableAaveSupplyPositions/index.ts

```diff
@@ -0,0 +1,142 @@
+  const [userReservesData, reservesData] = await publicClient.multicall({
+    contracts: [
+      {
+        abi: aaveUiPoolDataProviderAbi,
+        address: aaveUiPoolDataProviderContractAddress,
+        functionName: 'getUserReservesData',
+        args: [aavePoolAddressesProviderContractAddress, accountAddress],
+      },
+      {
+        abi: aaveUiPoolDataProviderAbi,
+        address: aaveUiPoolDataProviderContractAddress,
+        functionName: 'getReservesData',
+        args: [aavePoolAddressesProviderContractAddress],
+      },
+    ],
+  });
+
+  if (userReservesData.status === 'failure' || reservesData.status === 'failure') {
+    throw new VError({ type: 'unexpected', code: 'somethingWentWrong' });
+  }
+
+  // Map user aToken balances by underlying token address
+  const userATokenBalanceMapping: {
+    [tokenAddress: Address]: {
+      supplyBalanceMantissa: bigint;
+    };
+  } = {};
+
+  let userHasBorrowPosition = false;
+
+  userReservesData.result.forEach(data => {
+    if (typeof data === 'number') {
+      return;
+    }
+
+    data.forEach(({ underlyingAsset, scaledATokenBalance, scaledVariableDebt }) => {
+      if (scaledVariableDebt > 0n) {
+        userHasBorrowPosition = true;
+        return;
+      }
+
+      userATokenBalanceMapping[underlyingAsset.toLowerCase() as Address] = {
+        supplyBalanceMantissa: scaledATokenBalance,
+      };
+    });
+  });
+
+  // Return no importable positions if user has any borrow position
+  if (userHasBorrowPosition) {
+    return {
+      importableSupplyPositions: [],
+    };
+  }
+
+  const importableSupplyPositions: ImportableAaveSupplyPosition[] = [];
+
+  // Map reserves by underlying token address
+  reservesData.result.forEach(data => {
+    if ('marketReferenceCurrencyPriceInUsd' in data) {
+      return;
+    }
+
+    data.forEach(
+      ({ liquidityRate, liquidityIndex, underlyingAsset, aTokenAddress, availableLiquidity }) => {
+        const tokenAddress = underlyingAsset.toLowerCase() as Address;
+
+        // Skip assets for which user has no supply position
+        if (!Object.prototype.hasOwnProperty.call(userATokenBalanceMapping, tokenAddress)) {
+          return;
+        }
+
+        const userABalances = userATokenBalanceMapping[tokenAddress];
+
+        if (userABalances.supplyBalanceMantissa === 0n) {
+          return;
+        }
+
+        const userSupplyBalanceMantissa = BigInt(
+          new BigNumber(Number(userABalances.supplyBalanceMantissa))
+            .multipliedBy(Number(liquidityIndex))
+            .dividedBy(Number(RAY))
+            .toFixed(0, BigNumber.ROUND_DOWN),
+        );
+
+        // Skip positions for which pool doesn't have enough liquidity available to withdraw
+        if (availableLiquidity < userSupplyBalanceMantissa) {
+          return;
+        }
+
+        // Convert rate to APY
+        const ratePerSecond = Number(liquidityRate) / Number(RAY);
```

Comment:
Same here (`RAY` could maybe be a number right from its creation, no?)

Reply:
In this case, losing precision isn't really an issue because we're calculating an approximate APY.

I think `RAY` should remain a bigint, so that we can keep as much precision as possible when calculating the user supply balance in the block above.